### PR TITLE
ALTAPPS-584: iOS fix topic name doesn't fit to widget size

### PR DESCRIPTION
**Pull request by vladkash**
**YouTrack Issues**:
[#ALTAPPS-584](https://vyahhi.myjetbrains.com/youtrack/issue/ALTAPPS-584)

**Checklist**

_Before Code Review:_

- [x] Fields "Assignees, Labels, Milestone" are filled in the pull request;
- [x] All checks have been passed;
- [x] Changes have been checked locally.

**Description**
Added padding to button

### CODE REVIEW

File: iosHyperskillApp/iosHyperskillApp/Sources/Modules/Track/Views/DiscoverNext/TopicToDiscoverNextButtonView.swift [hunk 1]
```diff
@@ -10,7 +10,7 @@ extension TopicToDiscoverNextButtonView {
         var buttonStyle = OutlineButtonStyle(borderColor: .border, alignment: .leading, paddingEdgeSet: [])
 
         let learnNextBadgeVerticalPadding: CGFloat = 4
-        let learnNextBadgeVerticalOffsetRatio = -0.25
+        let learnNextBadgeVerticalOffsetRatio = -0.125
```
Comment (ivan-magda) [inline iosHyperskillApp/iosHyperskillApp/Sources/Modules/Track/Views/DiscoverNext/TopicToDiscoverNextButtonView.swift:13]: If the topic name will be longer -> badge will be outside of the widget (not centered)

Screenshot 2023-02-16 at 5 51 12 PM

You can add a view modifier to measure badge height from this [blog post](https://swiftuirecipes.com/blog/getting-size-of-a-view-in-swiftui).

Update badge rendering to this and all will be OK:
```swift
@ViewBuilder
private var learnNextBadge: some View {
    if isLearnNext {
        GeometryReader { geometry in
            Text(Strings.Track.TopicsToDiscoverNext.learnNextBadge)
                ...
                .measureSize { viewSize = $0 }
                .offset(
                    CGSize(
                        width: LayoutInsets.defaultInset,
                        height: -(viewSize.height / 2)
                    )
                )
        }
    }
}
```

Screenshot 2023-02-16 at 5 55 03 PM

File: iosHyperskillApp/iosHyperskillApp/Sources/Modules/Track/Views/DiscoverNext/TrackTopicsToDiscoverNextSkeletonListView.swift [hunk 2]
```diff
@@ -5,7 +5,7 @@ struct TrackTopicsToDiscoverNextSkeletonListView: View {
         VStack(spacing: LayoutInsets.defaultInset) {
             ForEach(0..<3) { _ in
                 SkeletonRoundedView()
-                    .frame(height: 44)
+                    .frame(height: 70)
```
Comment (ivan-magda) [inline iosHyperskillApp/iosHyperskillApp/Sources/Modules/Track/Views/DiscoverNext/TrackTopicsToDiscoverNextSkeletonListView.swift:8]:
```suggestion
                    .frame(height: 64)
```

File: iosHyperskillApp/iosHyperskillApp/Sources/Modules/Track/Views/DiscoverNext/TopicToDiscoverNextButtonView.swift [hunk 3]
```diff
@@ -77,7 +79,7 @@ struct TopicToDiscoverNextButtonView: View {
         )
         .buttonStyle(appearance.buttonStyle)
         // TODO: apply bounce animation to badge
-        .overlay(learnNextBadge)
+        .overlay(learnNextBadge, alignment: .topLeading)
```
Comment (ivan-magda) [inline iosHyperskillApp/iosHyperskillApp/Sources/Modules/Track/Views/DiscoverNext/TopicToDiscoverNextButtonView.swift:82]: 👍

File: iosHyperskillApp/iosHyperskillApp/Sources/Modules/Track/Views/DiscoverNext/TopicToDiscoverNextButtonView.swift [hunk 4]
```diff
@@ -98,21 +100,20 @@ struct TopicToDiscoverNextButtonView: View {
...
-                    .padding(.horizontal, LayoutInsets.smallInset)
-                    .padding(.vertical, appearance.learnNextBadgeVerticalPadding)
-                    .background(Color(ColorPalette.surface))
-                    .addBorder(color: Color(ColorPalette.primaryAlpha38))
-                    .offset(
-                        CGSize(
-                            width: LayoutInsets.defaultInset,
-                            height: geometry.size.height * appearance.learnNextBadgeVerticalOffsetRatio
-                        )
+            Text(Strings.Track.TopicsToDiscoverNext.learnNextBadge)
```
Comment (ivan-magda) [inline iosHyperskillApp/iosHyperskillApp/Sources/Modules/Track/Views/DiscoverNext/TopicToDiscoverNextButtonView.swift:103]: Cool without 2 `GeometryReader` as a suggested previously 👍

**Review by ivan-magda — APPROVED** [review]

File: iosHyperskillApp/iosHyperskillApp/Sources/Modules/Track/Views/DiscoverNext/TopicToDiscoverNextButtonView.swift (hunk 4 above)
Comment (vladkash) [reply iosHyperskillApp/iosHyperskillApp/Sources/Modules/Track/Views/DiscoverNext/TopicToDiscoverNextButtonView.swift:103]: Thank you for the idea!

**Pull request merged**